src/validation.cpp

```diff
@@ -3576,9 +3578,9 @@ bool BlockManager::LoadBlockIndex(
     // Calculate nChainWork
     std::vector<std::pair<int, CBlockIndex*> > vSortedByHeight;
     vSortedByHeight.reserve(m_block_index.size());
-    for (const std::pair<const uint256, CBlockIndex*>& item : m_block_index)
+    for (std::pair<const uint256, CBlockIndex>& item : m_block_index)
```

Comment:
> we're potentially introducing a lot of CBlockIndex value copying (vs. the pointers)

I don't think we're introducing value copying because we're iterating with `std::pair<...>&`, but perhaps there's something I'm missing? When I asked him about it, ryanofsky also pointed out that the const reference was actually more dangerous since it could have created copies through implicit conversion.